### ISSUE TITLE
incorrect handling of openTypeCodePageRanges custom parameter

### ISSUE THREAD

**Issue by BoldMonday** [googlei18n/glyphsLib#388]
Hi,

I am trying to convert a Glyphs file that only has `Thai` set in OS/2 CodePageRanges.

<img width="383" alt="thai" src="https://user-images.githubusercontent.com/13793769/43706845-c1c2d264-9966-11e8-8b6b-153941adb879.png">

But when using `glyphsLib.build_masters()` I get the following traceback:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/__init__.py", line 102, in build_masters
    minimize_glyphs_diffs=minimize_glyphs_diffs)
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/__init__.py", line 90, in to_designspace
    return builder.designspace
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/builders.py", line 201, in designspace
    ufos = list(self.masters)  # Make sure that the UFOs are built
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/builders.py", line 139, in masters
    self.to_ufo_font_attributes(self.family_name)
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/font.py", line 84, in to_ufo_font_attributes
    self.to_ufo_custom_params(ufo, font)
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/custom_params.py", line 619, in to_ufo_custom_params
    handler.to_ufo(glyphs_proxy, ufo_proxy)
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/custom_params.py", line 209, in to_ufo
    self._write_to_ufo(glyphs, ufo, ufo_value)
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/custom_params.py", line 247, in _write_to_ufo
    ufo.set_info_value(self.ufo_name, value)
  File "/code/glyphslib/lib/python3.5/site-packages/glyphsLib/builder/custom_params.py", line 145, in set_info_value
    setattr(self._owner.info, name, value)
  File "/code/glyphslib/lib/python3.5/site-packages/defcon/objects/info.py", line 38, in setter
    raise ValueError("Invalid value ({0}) for attribute {1}.".format(repr(value), name))
ValueError: Invalid value ([874]) for attribute openTypeOS2CodePageRanges.
```

**Comment by belluzj** [comment]
@BoldMonday Can you try renaming your custom parameter to `codePageRanges`? It should fix the problem.

The issue is that the code that translates from Glyphs codepage numbers to UFO codepage numbers assumes that if you name the custom parameter `codePageRanges` = it contains Glyphs values, but if you name it `openTypeOS2CodePageRanges` = it contains UFO values. From your example, I think that my assumption is wrong is and the code should be fixed to always do the conversion.

@schriftgestalt Are the codePageRange values in Glyphs always in the same format? Or does it change depending on the name of the parameter? (I can't remember where I got that assumption from)

https://github.com/googlei18n/glyphsLib/blob/a480b86452dd42dc2b1903d56629e9d26415fea7/Lib/glyphsLib/builder/custom_params.py#L375-L392

**Comment by anthrotype** [comment]
Hey Paul
You are using the UFO name of the fontinfo.plist parameter, that’s why it is expecting a list of numbers like the UFO spec says.
You want instead to use the Glyphs.app specific name for the same value which is `codePageRanges`. With the latter glyphsLib should be able to translate between the constant strings (e.g. Thai) to the respective code page ranges bit number.

**Comment by anthrotype** [comment]
Jany won

**Comment by moyogo** [comment]
Glyphs.app (2.4.1 or 2.5.2) gives the same dialog for `codePageRanges` or `openTypeCodePageRanges` and write the bits the same way (by CP number).

It seems glyphsLib should convert `openTypeCodePageRanges` the same way it converts `codePageRanges`.

**Comment by madig** [comment]
@schriftgestalt Why are codepage range bits 29, 30 and 31 not supported by Glyphs?

**Comment by anthrotype** [comment]
because makeotf doesn't support them, and glyphs.app writes a `OS/2` table override in the temporary features.fea that it sends to makeotf for compilation

**Comment by madig** [comment]
Should be fixed with #390. Please re-open if it isn't.
